_site/sw.js

```diff
@@ -107,22 +108,48 @@ function toggleTodo(todo) {
   };
 }
 
-async function redirect(path, isHtmx = false) {
-  return isHtmx 
-    ? new Response(null, { headers: new Headers({ "HX-Redirect": path }) })
-    : Response.redirect(path, 303);
+/**
+ * @param { string } path
+ * @param { boolean } [isSoftRedirect=false]
+ */
+async function redirect(path, isSoftRedirect = false) {
+  return isSoftRedirect
+  ? ServerSentEventGenerator.stream((stream) => {
+    stream.executeScript(`window.location = "${path}"`);
+  })
+  : Response.redirect(path, 303);
+}
+
+function ConfirmationDialog() {
```

Comment:
Use this component over the inlined version.

Or to keep it simple, remove this version